File: 0861-flipping-an-image/0861-flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        int n=image.length;
        
        for(int i=0;i<n;i++)
        {
            int left=0,right=n-1;
            while(left<=right)
            {
                if(image[i][left]==image[i][right])
                {
                    image[i][left]=image[i][right]=1-image[i][right];
                }
                left++;
                right--;
            }
        }
        return image;
    }
}

File: 0861-flipping-an-image/0861-flipping-an-image.py
class Solution:
    def flipAndInvertImage(self, image: List[List[int]]) -> List[List[int]]:
        n=len(image)
        for i in image:
            j=0
            k=n-1
            while(j<k):
                temp=i[j]
                i[j]=i[k]
                i[k]=temp
                k-=1
                j+=1
            for j in range(n):
                if i[j]:
                    i[j]=0
                else:
                    i[j]=1
        return image
        